outputs/aws/s3/s3.go

```diff
@@ -0,0 +1,119 @@
+package s3
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+
+	aws "github.com/falco-talon/falco-talon/internal/aws/client"
+	"github.com/falco-talon/falco-talon/internal/events"
+	"github.com/falco-talon/falco-talon/internal/rules"
+	"github.com/falco-talon/falco-talon/outputs/model"
+	"github.com/falco-talon/falco-talon/utils"
+)
+
+type Config struct {
+	Bucket string `mapstructure:"bucket" validate:"required"`
+	Prefix string `mapstructure:"prefix" validate:"required"`
+}
+
+func Output(output *rules.Output, data *model.Data) (utils.LogLine, error) {
+	parameters := output.GetParameters()
+	var config Config
+	err := utils.DecodeParams(parameters, &config)
+	if err != nil {
+		return utils.LogLine{
+			Objects: nil,
+			Error:   err.Error(),
+			Status:  "failure",
+		}, err
+	}
+
+	config.Prefix = strings.TrimSuffix(config.Prefix, "/")
+	config.Prefix = strings.TrimPrefix(config.Prefix, "/") + "/"
+
+	key := fmt.Sprintf("%v_%v_%v_%v", time.Now().Format("2006-01-02T15-04-05Z"), data.Namespace, data.Pod, strings.ReplaceAll(data.Name, "/", "_"))
+
+	objects := map[string]string{
+		"file":   data.Name,
+		"bucket": config.Bucket,
+		"prefix": config.Prefix,
+		"key":    key,
+	}
+
+	if err := putObject(config.Bucket, config.Prefix, key, *data); err != nil {
+		return utils.LogLine{
+			Objects: objects,
+			Error:   err.Error(),
+			Status:  "failure",
+		}, err
+	}
+
+	return utils.LogLine{
+		Objects: objects,
+		Output:  fmt.Sprintf("the file '%v' has been uploaded as the key '%v' to the bucket '%v'", data.Name, config.Prefix+key, config.Bucket),
+		Status:  "success",
+	}, nil
+}
+
+func CheckParameters(output *rules.Output) error {
+	parameters := output.GetParameters()
+
+	var config Config
+
+	err := utils.DecodeParams(parameters, &config)
+	if err != nil {
+		return err
+	}
+
+	err = utils.ValidateStruct(config)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func CheckBucketExist(output *rules.Output, _ *events.Event) error {
```

Comment:
function not being used currently, have you forgot to add it in the aws:s3 output section?

Reply:
After some research, writing an S3 Policy for the HeadBucket request is tricky, maybe even not possible, so I preferred to remove the check of the bucket for AWS S3. The client will receive a 404 when it will attempt to put the object, the behavior is same then.